lib/lifecycles/bump.js

```diff
@@ -20,16 +20,45 @@ async function Bump (args, version) {
   configsToUpdate = {}
 
   if (args.skip.bump) return version
+
+  if (args.releaseAs && !(['major', 'minor', 'patch'].includes(args.releaseAs.toLowerCase()) || semver.valid(args.releaseAs))) {
+    throw Error("releaseAs must be one of 'major', 'minor' or 'patch', or a valid semvar version.")
+  }
+
   let newVersion = version
   await runLifecycleScript(args, 'prerelease')
   const stdout = await runLifecycleScript(args, 'prebump')
-  if (stdout && stdout.trim().length) args.releaseAs = stdout.trim()
-  const release = await bumpVersion(args.releaseAs, version, args)
+  if (stdout?.trim().length) {
```

Comment:
Optional: I think the previous version of this if was clearer, but that might be personal preference.

Reply:
SonarLint yelled at me to change that one, which is the only reason I did. I'm easy with it either way, though they're opinionated in favor of the null chaining operator.